packages/edit-site/src/components/layout/index.js

```diff
@@ -247,21 +256,13 @@ export default function Layout() {
 
 				<div className="edit-site-layout__content">
 					<AnimatePresence initial={ false }>
-						{
+						{ showSidebar && (
```

Comment:
@kevin940726 mentioned to me that the issue with completely unmounting the sidebar is that it contains hooks that sync the routing state, and while it's unmounted those hooks aren't running, so some navigation doesn't work as expected. This is the reason it was changed in #51558.

@youknowriad Wondering if you have any suggestions for solving that in a different way?

Reply:
Specifically, here's a bug I found while testing on mobile:

https://github.com/WordPress/gutenberg/assets/7753001/4a274290-19c0-49ea-9647-bc964eb4077f

The previous sidebar still exists but hidden if we use the browser's back button to go back from the edit mode. It has a higher `z-index` value too so it intercepts the click event (the add new pattern dropdown shouldn't be opened).

There might be a better and more elegant solution for this though! I didn't have time so I settled with the CSS hack 🤦 .

Reply:
Also related is the issue found [here](https://github.com/WordPress/gutenberg/pull/50565#issuecomment-1564153060) that actually existed before that PR and is potentially related to AnimatePresence not unmounting properly. The added complication is we now need the sidebar mounted on mobile for routing reasons.

Reply:
> it contains hooks that sync the routing state

Can you clarify which hooks we're talking about here?

Reply:
@kevin940726 Do you know the which specific code might be causing that issue?

Reply:
I think it's `useSyncPathWithURL` which is called inside the sidebar. I'm not entirely sure if we need it here though, this whole thing is a bit out of my league.

Reply:
Can't we move `useSyncPathWithURL` to the layout component instead?

Reply:
IIRC, that hook relies on `useNavigator` which relies on the Navigator context, so we probably have to move the whole `NavigatorProvider` too. Last time I tried it failed on a weird error which I can't recall. To minimize the scope of the PR, fixing the CSS seems like an easier solution for now.